docs/src/components/ReactExample/components/ViewportsRuler.tsx

```diff
@@ -51,62 +50,70 @@ const StyledCell = styled(({ className, children, onClick, onMouseEnter }) => (
   const { isLargeMobile } = useMediaQuery();
 
-  const setRuler = React.useCallback(() => {
-    const currentViews = Object.entries(QUERIES)
-      .filter(([, value]) => value <= windowWidth)
-      .concat([["fullWidth", windowWidth]])
-      .reverse();
-
-    setActiveIdx(currentViews.length - 1);
-
-    setViewports(currentViews);
-  }, [windowWidth, setViewports]);
-
-  const handleResize = React.useCallback(() => {
-    onChangeSize(QUERIES.smallMobile);
-    setRuler();
-    setVisibleValue("smallMobile (320)");
-    if (ref.current) setWindowWidth(ref.current?.clientWidth);
-  }, [onChangeSize, setRuler]);
-
   React.useEffect(() => {
-    handleResize();
+    const newViewports = Object.keys(QUERIES).filter(
+      viewport =>
+        typeof containerWidth !== "number" ||
+        typeof QUERIES[viewport] === "string" ||
+        QUERIES[viewport] < containerWidth,
+    );
 
-    window.addEventListener("resize", handleResize);
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
-  }, [handleResize, windowWidth]);
+    setViewports(prevViewports =>
+      prevViewports.length === newViewports.length ? prevViewports : newViewports,
+    );
+    setActiveViewport(prevActive =>
+      newViewports.includes(prevActive) ? prevActive : newViewports[newViewports.length - 1],
+    );
+  }, [containerWidth]);
 
-  const handleClick = (size: number, idx: number) => {
-    onChangeSize(size);
-    setActiveIdx(idx);
-  };
+  React.useEffect(() => {
+    onChangeSize(QUERIES[activeViewport]);
+  }, [activeViewport, onChangeSize]);
 
-  return isLargeMobile ? (
-    <StyledViewports ref={ref}>
-      {viewports.map(([name, value], i) => (
-        <StyledCell
-          active={activeIdx === i}
-          key={name}
-          width={value}
-          index={i}
-          onMouseEnter={() => setVisibleValue(`${name} (${value})`)}
-          onClick={() => handleClick(value, i)}
-        >
-          <span>{visibleValue}</span>
-        </StyledCell>
-      ))}
-    </StyledViewports>
-  ) : null;
+  return (
+    <div ref={containerRef}>
+      {isLargeMobile && (
+        <StyledViewports>
+          {[...viewports].reverse().map(viewport => (
+            <StyledCell
+              key={viewport}
+              aria-label={`${viewport} (${QUERIES[viewport]})`}
+              $active={targetViewport ? viewport === targetViewport : viewport === activeViewport}
+              $width={QUERIES[viewport]}
+              onMouseEnter={() => setTargetViewport(viewport)}
+              onMouseLeave={() => setTargetViewport(null)}
+              onClick={() => setActiveViewport(viewport)}
+            />
+          ))}
+          <StyledLabel>
+            <Text as="div">
+              {targetViewport
+                ? `${targetViewport} (${QUERIES[targetViewport]})`
+                : `${activeViewport} (${QUERIES[activeViewport]})`}
```

Comment:
in one of the latests PRs i added pixels. Could you add that? so it's not just 320, but `320px` and etc.